controllers/provisioning_controller.go

```diff
@@ -141,6 +143,11 @@ func (r *ProvisioningReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 		// Temporarily not requeuing request
 		return ctrl.Result{}, nil
 	}
+	//Create Secrets needed for Metal3 deployment
+	if err := provisioning.CreateMetal3PasswordSecrets(r.kubeClient.CoreV1(), ComponentNamespace); err != nil {
+		r.Log.V(1).Info("Unable to create one or more Secrets")
+		return ctrl.Result{}, err
```

Comment:
As per recent discussion on logging style, I think this should be:
`return ctrl.Result{}, errors.Wrap(err, "unable to create one or more Secrets")}`

Reply:
Interesting because I did try to follow what https://github.com/openshift/cluster-baremetal-operator/pull/24 is doing.

Do I really need to wrap this error again? What we are getting from CreateMetal3PasswordSecrets() is a specific error msg about which password creation failed with the error from the underlying call.

Reply:
In a concurrent system like this, we want each event to have 1 log message and for that message to include all of the information someone would need to understand what's going on. So we almost never want to log and then return an error that the caller will also log, because then we end up with 2 log messages on separate lines. The log aggregation tool won't know they are related, so information will be harder to find.

The message in the log line here, "Unable to create one or more secrets," could be made more helpful by adding context about the secret(s) that couldn't be created -- names, purpose, etc. Splitting that function into multiple calls will make it easier to provide that context, because the message can say "could not create foo secret".

Reply:
The error message returned by CreateMetal3PasswordSecrets() tells exactly which secret had the error and that would definitely be part of the err returned here. 
I'll remove the log message to avoid logging and then returning an error.